jsonschema/json.go

```diff
@@ -33,3 +35,25 @@ type Definition struct {
 	// Items specifies which data type an array contains, if the schema type is Array.
 	Items *Definition `json:"items,omitempty"`
 }
+
+func (d *Definition) MarshalJSON() ([]byte, error) {
+	d.initializeProperties()
+	return json.Marshal(*d)
+}
+
+func (d *Definition) initializeProperties() {
+	if d.Properties == nil {
+		d.Properties = make(map[string]Definition)
+		return
+	}
+
+	for k, v := range d.Properties {
+		if v.Properties == nil {
+			v.Properties = make(map[string]Definition)
+			d.Properties[k] = v
+		} else {
+			v.initializeProperties()
+			d.Properties[k] = v
+		}
```

Comment:
nit:

```go
if v.Properties == nil {
	v.Properties = make(map[string]Definition)
} else {
	v.initializeProperties()
}

d.Properties[k] = v
```

Reply:
@sashabaranov Oops! Thanks for the review.